Sources/GRPC/GRPCChannel/GRPCChannelBuilder.swift

```diff
@@ -0,0 +1,188 @@
+        tls: self.maybeTLS,
+        connectionBackoff: self.connectionBackoffIsEnabled ? self.connectionBackoff : nil
+      )
+      return ClientConnection(configuration: configuration)
+    }
+  }
+}
+
+extension ClientConnection.Builder {
+  public class Secure: ClientConnection.Builder {
+    internal var tls = ClientConnection.Configuration.TLS()
+    override internal var maybeTLS: ClientConnection.Configuration.TLS? {
+      return self.tls
+    }
+  }
+}
+
+extension ClientConnection.Builder {
+  /// Sets the initial connection backoff. That is, the initial time to wait before re-attempting to
+  /// establish a connection. Jitter will *not* be applied to the initial backoff. Defaults to
+  /// 1 second if not set.
+  @discardableResult
+  public func withConnectionBackoff(initial amount: TimeAmount) -> Self {
+    self.connectionBackoff.initialBackoff = Double(amount.nanoseconds) / Double(1_000_000_000)
+    return self
+  }
+
+  /// Set the maximum connection backoff. That is, the maximum amount of time to wait before
+  /// re-attempting to establish a connection. Note that this time amount represents the maximum
+  /// backoff *before* jitter is applied. Defaults to 120 seconds if not set.
+  @discardableResult
+  public func withConnectionBackoff(maximum amount: TimeAmount) -> Self {
+    self.connectionBackoff.maximumBackoff = Double(amount.nanoseconds) / Double(1_000_000_000)
+    return self
+  }
+
+  /// Backoff is 'jittered' to randomise the amount of time to wait before re-attempting to
+  /// establish a connection. The jittered backoff will be no more than `jitter ⨯ unjitteredBackoff`
+  /// from `unjitteredBackoff`. Defaults to 0.2 if not set.
+  ///
+  /// - Precondition: `0 <= jitter <= 1`
+  @discardableResult
+  public func withConnectionBackoff(jitter: Double) -> Self {
+    self.connectionBackoff.jitter = jitter
+    return self
+  }
+
+  /// The multiplier for scaling the unjittered backoff between attempts to establish a connection.
+  /// Defaults to 1.6 if not set.
+  @discardableResult
+  public func withConnectionBackoff(multiplier: Double) -> Self {
+    self.connectionBackoff.multiplier = multiplier
+    return self
+  }
+
+  /// The minimum timeout to use when attempting to establishing a connection. The connection
+  /// timeout for each attempt is the larger of the jittered backoff and the minimum connection
+  /// timeout. Defaults to 20 seconds if not set.
+  @discardableResult
+  public func withConnectionTimeout(minimum amount: TimeAmount) -> Self {
+    self.connectionBackoff.minimumConnectionTimeout = Double(amount.nanoseconds) / Double(1_000_000_000)
+    return self
+  }
+
+  /// Sets the initial and maximum backoff to given amount. Disables jitter and sets the backoff
+  /// multiplier to 1.0.
+  @discardableResult
+  public func withConnectionBackoff(fixed amount: TimeAmount) -> Self {
+    let seconds = Double(amount.nanoseconds) / Double(1_000_000_000)
```

Comment:
Constant 1_000_000_000 is used a couple of times. Perhaps make attached it to `Double` to reduce potential error? (E.g. Double.Nanosecond, or just make it a constant somewhere)